x/supplier/keeper/msg_server_stake_supplier_test.go

```diff
@@ -360,20 +364,67 @@ func stakeSupplierForServicesMsg(
 		})
 	}
 
-	return &types.MsgStakeSupplier{
+	return &suppliertypes.MsgStakeSupplier{
 		Signer:          ownerAddr,
 		OwnerAddress:    ownerAddr,
 		OperatorAddress: operatorAddr,
-		Stake:           &sdk.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(amount)},
+		Stake:           &cosmostypes.Coin{Denom: volatile.DenomuPOKT, Amount: math.NewInt(amount)},
 		Services:        services,
 	}
 }
 
 // setStakeMsgSigner sets the signer of the given MsgStakeSupplier to the given address
 func setStakeMsgSigner(
-	msg *types.MsgStakeSupplier,
+	msg *suppliertypes.MsgStakeSupplier,
 	signer string,
-) *types.MsgStakeSupplier {
+) *suppliertypes.MsgStakeSupplier {
 	msg.Signer = signer
 	return msg
 }
+
+func TestMsgServer_StakeSupplier_FailBelowMinStake(t *testing.T) {
+	k, ctx := keepertest.SupplierKeeper(t)
+	srv := keeper.NewMsgServerImpl(*k.Keeper)
+
+	addr := sample.AccAddress()
+	supplierStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 100)
+	minStake := supplierStake.AddAmount(math.NewInt(1))
+	expectedErr := suppliertypes.ErrSupplierInvalidStake.Wrapf("supplier with owner %q must stake at least %s", addr, minStake)
+
+	// Set the minimum stake to be greater than the supplier stake.
+	params := k.Keeper.GetParams(ctx)
+	params.MinStake = &minStake
+	err := k.SetParams(ctx, params)
+	require.NoError(t, err)
+
+	// Prepare the application.
+	stakeMsg := &suppliertypes.MsgStakeSupplier{
+		Signer:          addr,
+		OwnerAddress:    addr,
+		OperatorAddress: addr,
+		Stake:           &supplierStake,
+		Services: []*sharedtypes.SupplierServiceConfig{
+			{
+				ServiceId: "svcId",
+				Endpoints: []*sharedtypes.SupplierEndpoint{
+					{
+						Url:     "http://test.example:8080",
+						RpcType: sharedtypes.RPCType_JSON_RPC,
+					},
+				},
+				RevShare: []*sharedtypes.ServiceRevenueShare{
+					{
+						Address:            addr,
+						RevSharePercentage: 100,
+					},
+				},
+			},
+		},
+	}
+
+	// Attempt to stake the supplier & verify that the supplier does NOT exist.
+	_, err = srv.StakeSupplier(ctx, stakeMsg)
```

Comment:
Is it worth testing an upstaking (below min stake) of a supplier that already have its stake below minimum?

Is it even possible (according to the stake keeper method it is)

Reply:
How do you see it being possible to get to a state where a supplier has below the min stake?

1. The `StakeApplication()` handler prevents a new supplier from staking below the minimum.
2. There is no way to partially unstake (all or nothing) any actor.

The only current pathway would be via supplier slashing, which was in progress at the time. The supplier SHOULD be unstaked if their stake drops below the min as a result of slashing; this will be added in a subsequent PR.